### PR TITLE
Feature vehicles stats

### PR DESCRIPTION
**Pull request by peter-budo**
@karllindmark 

Vehicle stats available. Not much to say about it.
- Weapon stats JSON handling classes been in wrong package so moved
- Created UI and functional stuff for vehicles
- Fixed some of already existing resources and added new vehicle group string map
- Corrected imports, seems like they got messed up little for some reason

I know that groupVehicle class inside VehicleStatistics https://github.com/karllindmark/com.ninetwozero.battlelog/blob/846bea2e7ea7952cde6b7d0f721a9b95fbf44ebe/core/src/main/java/com/ninetwozero/bf4intel/json/stats/vehicles/VehicleStatistics.java#L22 is rather robust but it does quick job of sorting all vehicles in groups and setting up star count, kills and progression

### CODE REVIEW

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/factories/UrlFactory.java [hunk 1]
```diff
@@ -26,6 +26,10 @@ public static URL buildWeaponStatsURL(final int soldierId, final int platformId)
         return createURL(String.format("warsawWeaponsPopulateStats/%d/%d/stats/", soldierId, platformId));
     }
 
+    public static URL buildVehicleStatsURL(final int soldierId, final int platformId) {
```
Comment (karllindmark) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/factories/UrlFactory.java:29]: Should we keep soldierId as int or String? Do we get any benefits of keeping it as int? :-)

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/resources/maps/vehicles/VehicleImageMap.java [hunk 2]
```diff
@@ -50,14 +50,16 @@
             put("WARSAW_ID_P_VNAME_Z9", R.drawable.vehicle_z9_haitun);
             put("WARSAW_ID_P_VNAME_ZBD09", R.drawable.vehicle_zbd09);
             put("WARSAW_ID_P_VNAME_ZFB05", R.drawable.vehicle_zfb05);
-// Not available yet            put("WARSAW_ID_P_XP0_VNAME_DPV", R.drawable.vehicle_xp0_dpv);
```
Comment (karllindmark) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/resources/maps/vehicles/VehicleImageMap.java:53]: :+1:

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/resources/maps/vehicles/VehiclesGroupStringMap.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package com.ninetwozero.bf4intel.resources.maps.vehicles;
+
+import com.ninetwozero.bf4intel.R;
+import java.util.HashMap;
+import java.util.Map;
+
+public class VehiclesGroupStringMap {
+
+    private static final Map<String, Integer> map = new HashMap<String, Integer>() {
```
Comment (karllindmark) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/resources/maps/vehicles/VehiclesGroupStringMap.java:9]: Splendid - I'll be using this on my end when it's merged :))

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/stats/vehicles/VehicleStatsFragment.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+import com.ninetwozero.bf4intel.network.SimpleGetRequest;
+import com.ninetwozero.bf4intel.utils.Result;
+
+import java.util.List;
+
+public class VehicleStatsFragment extends BaseLoadingListFragment {
+
+    private static final int ID_LOADER = 2200;
+    private ListView listView;
+    private VehicleStatistics vs;
```
Comment (karllindmark) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/stats/vehicles/VehicleStatsFragment.java]: Do we need to save `vs` and `adapter` at the moment? Could these be changed to an inline `final VehicleStatistics vehicleStatistics`/`final VehicleStatsAdapter adapter` inside the method instead? Just a thought. :)

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/stats/weapons/WeaponStatsAdapter.java [hunk 5]
```diff
@@ -29,17 +29,17 @@ public long getItemId(int position) {
     @Override
     public View getView(int position, View view, ViewGroup parent) {
         if (view == null) {
-            view = LayoutInflater.from(context).inflate(R.layout.list_item_weapon, parent, false);
+            view = LayoutInflater.from(context).inflate(R.layout.list_stats_item, parent, false);
```
Comment (karllindmark) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/stats/weapons/WeaponStatsAdapter.java:32]: Reusability :+1:

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/stats/vehicles/VehicleStatsFragment.java (hunk 4 above)
Comment (peter-budo) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/stats/vehicles/VehicleStatsFragment.java]: `vs` is forgotten artefact, however I believe `adapter` will be need it as soon we start to apply view states

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/factories/UrlFactory.java (hunk 1 above)
Comment (peter-budo) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/factories/UrlFactory.java:29]: Not decided yet for 100% so for now sticking with keeping `soldier Id` what truly is integer

File: core/src/main/java/com/ninetwozero/bf4intel/json/stats/vehicles/VehicleStatistics.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+    public Map<String, VehicleServiceStars> getUnlockProgressions() {
+        return unlockProgressions;
+    }
+
+    public List<VehicleStats> getVehiclesList() {
+        return vehiclesList;
+    }
+
+    public List<GroupedVehicleStats> groupVehicles() {
```
Comment (karllindmark) [inline core/src/main/java/com/ninetwozero/bf4intel/json/stats/vehicles/VehicleStatistics.java]: `fetchGroupedVehicles()`?

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/stats/vehicles/VehicleStatsFragment.java (hunk 4 above)
Comment (karllindmark) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/ui/stats/vehicles/VehicleStatsFragment.java]: We should probably remove `vs`, but `adapter` is OK to keep imo. :+1:

**Comment by karllindmark** [comment]
Not sure if anything breaking due to changes here, but I can't switch from weapons --> vehicles --> weapons - I get a blank screen when going back to weapons.

If we _add_ the unlocks-impl PR first, we can backport the ViewPager functionality into this one as well. It should probably make a difference (if not only enabling the swipe).

:moneybag:

**Comment by karllindmark** [comment]
I noticed that the images are still low-res - do we want to update them now or in the future?

**Comment by peter-budo** [comment]
Leave images for now

**Comment by peter-budo** [comment]
@karllindmark give it a go...

**Comment by karllindmark** [comment]
Will be home in a bit, will check then :-)

**Comment by karllindmark** [comment]
Nice catch @peter-budo! Anything else we need to address here, or merge+move on to reports?

**Pull request merged**